[BUG] koalas.sql fails with java.lang.ArrayIndexOutOfBoundsException

**Describe the bug**
A clear and concise description of what the bug is.

Koalas.sql fails with java.lang.ArrayIndexOutOfBoundsException.
Sample stacktrace is:

```
Caused by: java.lang.ArrayIndexOutOfBoundsException: Index 0 out of bounds for length 0
	at ai.rapids.cudf.Table.<init>(Table.java:62)
	at com.nvidia.spark.rapids.GpuColumnVector.from(GpuColumnVector.java:498)
	at com.nvidia.spark.rapids.GpuWindowExpression.$anonfun$evaluateRowBasedWindowExpression$1(GpuWindowExpression.scala:188)
	at com.nvidia.spark.rapids.Arm.withResource(Arm.scala:28)
	at com.nvidia.spark.rapids.Arm.withResource$(Arm.scala:26)
	at com.nvidia.spark.rapids.GpuWindowExpression.withResource(GpuWindowExpression.scala:132)
	at com.nvidia.spark.rapids.GpuWindowExpression.evaluateRowBasedWindowExpression(GpuWindowExpression.scala:187)
	at com.nvidia.spark.rapids.GpuWindowExpression.columnarEval(GpuWindowExpression.scala:167)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
	at com.nvidia.spark.rapids.GpuAlias.columnarEval(namedExpressions.scala:93)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$ReallyAGpuExpression.columnarEval(implicits.scala:35)
	at com.nvidia.spark.rapids.GpuProjectExec$.$anonfun$project$1(basicPhysicalOperators.scala:50)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1(implicits.scala:161)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.$anonfun$safeMap$1$adapted(implicits.scala:158)
	at scala.collection.immutable.Stream.foreach(Stream.scala:533)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$MapsSafely.safeMap(implicits.scala:158)
	at com.nvidia.spark.rapids.RapidsPluginImplicits$AutoCloseableProducingSeq.safeMap(implicits.scala:193)
	at com.nvidia.spark.rapids.GpuProjectExec$.project(basicPhysicalOperators.scala:49)
	at com.nvidia.spark.rapids.GpuProjectExec$.projectAndClose(basicPhysicalOperators.scala:40)
	at com.nvidia.spark.rapids.GpuWindowExec.$anonfun$doExecuteColumnar$1(GpuWindowExec.scala:149)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at scala.collection.Iterator$$anon$10.next(Iterator.scala:459)
	at com.nvidia.spark.rapids.GpuBaseLimitExec$$anon$1.next(limit.scala:67)
	at com.nvidia.spark.rapids.GpuBaseLimitExec$$anon$1.next(limit.scala:61)
	at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExec$$anon$1.partNextBatch(GpuShuffleExchangeExec.scala:206)
	at org.apache.spark.sql.rapids.execution.GpuShuffleExchangeExec$$anon$1.hasNext(GpuShuffleExchangeExec.scala:222)
	at org.apache.spark.shuffle.sort.BypassMergeSortShuffleWriter.write(BypassMergeSortShuffleWriter.java:132)
	at org.apache.spark.shuffle.ShuffleWriteProcessor.write(ShuffleWriteProcessor.scala:59)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:99)
	at org.apache.spark.scheduler.ShuffleMapTask.runTask(ShuffleMapTask.scala:52)
	at org.apache.spark.scheduler.Task.run(Task.scala:131)
	at org.apache.spark.executor.Executor$TaskRunner.$anonfun$run$3(Executor.scala:497)
	at org.apache.spark.util.Utils$.tryWithSafeFinally(Utils.scala:1439)
	at org.apache.spark.executor.Executor$TaskRunner.run(Executor.scala:500)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	... 1 more
```



**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

Minimum reproduce using .sql
```
import databricks.koalas as ks

kdf = ks.DataFrame({'a': [1, 2, 3],
                    'b': [4, 5, 6]})
ks.sql("SELECT * FROM {kdf} WHERE a > 1")
```

Some other Koalas functions also fail with similar stacktrace, 2nd example:
```
import numpy as np
import databricks.koalas as ks

kdf = ks.DataFrame({'A': np.random.rand(5),
                    'B': np.random.rand(5)})
                    
def square(x) -> ks.Series[np.float64]:
    return x ** 2
kdf.apply(square)
```

**Expected behavior**
A clear and concise description of what you expected to happen.

Above 2 examples should work fine such as in CPU mode.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

Spark standalone cluster 3.1.1
Rapids accelerator 0.4.1
Koalas 1.7

**Additional context**
Add any other context about the problem here.

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -187,7 +187,7 @@ case class GpuWindowExpression(windowFunction: Expression, windowSpec: GpuWindow
     val aggColumn = withResource(GpuProjectExec.project(cb, boundRowProjectList)) { projected =>
 
       // in case boundRowProjectList is empty
-      val finalCb = if (boundRowProjectList.length > 0) projected else cb
+      val finalCb = if (boundRowProjectList.nonEmpty) projected else cb
 
       withResource(GpuColumnVector.from(finalCb)) { table =>
         val bases = GpuColumnVector.extractBases(finalCb).zipWithIndex

--- a/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala
+++ b/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala
@@ -185,8 +185,12 @@ case class GpuWindowExpression(windowFunction: Expression, windowSpec: GpuWindow
     val totalExtraColumns = numGroupingColumns
 
     val aggColumn = withResource(GpuProjectExec.project(cb, boundRowProjectList)) { projected =>
-      withResource(GpuColumnVector.from(projected)) { table =>
-        val bases = GpuColumnVector.extractBases(projected).zipWithIndex
+
+      // in case boundRowProjectList is empty
+      val finalCb = if (boundRowProjectList.length > 0) projected else cb
+
+      withResource(GpuColumnVector.from(finalCb)) { table =>
+        val bases = GpuColumnVector.extractBases(finalCb).zipWithIndex
             .slice(totalExtraColumns, boundRowProjectList.length)
 
         val agg = windowFunc.windowAggregation(bases)

--- a/tests/src/test/scala/com/nvidia/spark/rapids/WindowFunctionSuite.scala
+++ b/tests/src/test/scala/com/nvidia/spark/rapids/WindowFunctionSuite.scala
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, NVIDIA CORPORATION.
+ * Copyright (c) 2020-2021, NVIDIA CORPORATION.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.

--- a/tests/src/test/scala/com/nvidia/spark/rapids/WindowFunctionSuite.scala
+++ b/tests/src/test/scala/com/nvidia/spark/rapids/WindowFunctionSuite.scala
@@ -166,6 +166,15 @@ class WindowFunctionSuite extends SparkQueryCompareTestSuite {
     rowNumberAggregationTesterForDecimal(rowsWindow, scale = 2)
   }
 
+  testSparkResultsAreEqual(
+    "[Window] [ROWS] [UNBOUNDED PRECEDING, CURRENT ROW] [ROW_NUMBER] [WITHOUT PARTITIONBY]",
+    windowTestDfOrc) {
+    val rowsWindow = Window.orderBy("dateLong")
+      .rowsBetween(Window.unboundedPreceding, 0)
+    rowNumberAggregationTester(rowsWindow)
+    rowNumberAggregationTesterForDecimal(rowsWindow, scale = 2)
+  }
+
   testSparkResultsAreEqual("[Window] [ROWS] [UNBOUNDED PRECEDING, UNBOUNDED FOLLOWING] ",
       windowTestDfOrc) {
     val rowsWindow = Window.partitionBy("uid")

